engine/engine/content/builtins/scripts/edn.lua

```diff
@@ -14,34 +20,15 @@ local escape_char_map = {
   ["\t" ] = "\\t",
 }
 
-local function raw_tostring(val)
-  local old_mt = debug.getmetatable(val)
-  debug.setmetatable(val, nil)
-  local raw_str = tostring(val)
-  debug.setmetatable(val, old_mt)
-  return raw_str
-end
-
-local function has_custom_tostring(val)
-  local mt = debug.getmetatable(val)
-  return mt ~= nil and mt.__tostring ~= nil
-end
-
-local function get_prefixed_addr(prefix, val)
-  local raw_string = raw_tostring(val)
-  local b, e = string.find(raw_string, prefix)
-  if e ~= nil then
-    return string.sub(raw_string, e + 1)
-  else
-    return ""
-  end
-end
-
 local function str(val)
-  if getmetatable(val) == NodeProxy then
-    return tostring(val) .. get_prefixed_addr("userdata: ", val)
-  elseif type(val) == "table" and has_custom_tostring(val) then
-    return tostring(val) .. " " .. get_prefixed_addr("table: ", val)
```

Comment:
here we take metatable twice almost in 100% cases (`getmetatable()`, `has_custom_tostring()`)
So it's better to reduce count of such calls